apps/api/src/exception-filter.ts

```diff
@@ -96,6 +80,41 @@ export class AllExceptionsFilter implements ExceptionFilter {
       message: `Internal server error, contact support and provide them with the errorId`,
     };
   }
+  private handleCommandValidation(exception: CommandValidationException): ResponseMetadata {
+    const DO_NOT_TRACK_CLASSES = [ProcessBulkTriggerCommand.name];
+
+    let uuid: string | undefined;
+    if (!DO_NOT_TRACK_CLASSES.includes(exception.className)) {
+      uuid = this.getUuid(exception);
+      this.logError(uuid, exception);
+    }
+
+    return {
+      message: {
+        message: exception.message,
+        cause: exception.constraintsViolated,
```

Comment:
can you provide an example of the error we had previously and what we are getting now? I don’t recall seeing cause in the past, but it’s possible I haven’t interacted with command errors recently.

Reply:
I've refactored the structure of the validation in a similar way to what you did, but before it will just shoot out a string list in the message now we have a more structured response, you can see the old structure in the tests changed 
the actual value in cause now is: 
    public constraintsViolated: Record<string, ConstraintValidation>,

class ConstraintValidation {
  messages: string[];
  value: any;
}

Reply:
Could you elaborate on why you chose this more structured object other than `"message": []`. is there a common practice to do it in this way to the API client can easier to consume it?

In addition, i am not sure i fully understand what the structure and what `string` stands for in:
`Record<string, ConstraintValidation>`
because if it is the first layer variables, i do not understand how this object is supporting nested validation. in another comment, you provided the following example:

> {
>     status: 400,
>     message: "Failed Validation",
>     cause: {name: {value: "", messages:['Name cannot be empty']
> }

But how will it represent an error if we have an issue with the following structure as an example `workflow.steps[3].template.name`

Reply:
it will show 
{
status: 400 
message: "Failed Validation" 
cause :{
"workflow.steps[3].template.name": "name cannot be empty"
}
}
API client shows a body of the error, there is no standard practice here, it needs to be a clear error, also we don't automatically consume it, it just needs to be clear and readable, just dumping a json in the message doesn't make it more readable